doc/manual/src/design/store/store.md

```diff
@@ -6,22 +6,21 @@ Every store also has a "store directory path", which is a path prefix used for v
 There are many types of stores, but all of them at least respect this model.
 Some however offer additional functionality.
 
-## A Rosetta stone for the Nix store.
+## A [Rosetta stone](https://en.m.wikipedia.org/wiki/Rosetta_Stone) for build system terminology
 
-The design of Nix is comparable to other build systems, even programming languages in general.
-Here is a rough [Rosetta stone](https://en.m.wikipedia.org/wiki/Rosetta_Stone) for build system terminology.
-If you are familiar with one of these columns, this might help the following sections make more sense.
+The Nix store's design is comparable to other build systems.
+Usage of terms is, for historic reasons, not entirely consistent within the Nix ecosystem, and still subject to slow change.
 
-generic build system | Nix | Bazel | Build Systems à la Carte | lazy programming language
--- | -- | -- | -- | --
-data (build input, build result) | component | file (source, target) | value | value
-build plan | derivation graph | action graph | `Tasks` | thunk
-build step | derivation | rule | `Task` | thunk
-build instructions | builder | (depends on action type) | `Task` | function
-build | build | build | `Build` applied to arguments | evaluation
-persistence layer | store | file system | `Store` | heap
+The following translation table points out similarities and equivalent terms, to help clarify their meaning and inform consistent use in the future.
 
-(n.b. Bazel terms gotten from https://docs.bazel.build/versions/main/glossary.html.)
+generic build system | Nix | [Bazel](https://bazel.build/start/bazel-intro) | [Build Systems à la Carte](https://www.microsoft.com/en-us/research/uploads/prod/2018/03/build-systems.pdf) | programming language
+-- | -- | -- | -- | --
+data (build input, build result) | component | [artifact](https://bazel.build/reference/glossary#artifact) | value | value
+build plan | derivation | [rule](https://bazel.build/reference/glossary#rule) | `Task` | [thunk](https://en.m.wikipedia.org/wiki/Thunk)
+build graph | derivation graph | [action graph](https://bazel.build/reference/glossary#action-graph), [build graph](https://bazel.build/reference/glossary#build-graph) | `Tasks` | [call graph](https://en.m.wikipedia.org/wiki/Call_graph)
+build instructions | builder | [action](https://bazel.build/reference/glossary#action) | `Task` | function
+build | realisation | build | application of `Build` | evaluation
```

Comment:
@edolstra Is right we should not use "realisation" because it is now used for something else with CA derivations.

Reply:
I wanted to open this particular discussion here, because we still have that terminology flying around.
1. CA derivations are still experimental and can change usage of that term, right @thufschmitt?
2. Even if "realisation" is overloaded, that is what most of the material still calls building.

The latter is absolutely fine, as is calling generic software stuff "components", because Nix is abstract for good reason, and we're currently just trying to shoehorn it into build systems.

Another question I try to touch on in the commit message is how we want to shape terminology in the future. We could rename the Nix column "Nix 2" and add another for "Nix 3" where we have _realisation->build_ and _component -> data (build input, build result)_.

Or we drop the "generic build system" column altogether and just stick to derivation everywhere. It should be enough to introduce it in the overview with Eelco's apt original wording that it is used to _derive_ build results from build inputs using instructions encoded in the builder.

A strong argument for the latter is that that term is all over the place and we won't change that by adding a chapter to the manual, even if there was a good reason to do so (which I don't see).